### PR TITLE
dumpling: New terminology for MySQL

### LINKED ISSUE #53082: dumpling: Support for MySQL 8.4.0
## Enhancement

```
dumpling/export/metadata.go:		buffer.WriteString("SHOW MASTER STATUS:")
dumpling/export/metadata.go:	// mysql> SHOW MASTER STATUS;
dumpling/export/metadata.go:	// mysql> SHOW MASTER STATUS;
dumpling/export/metadata.go:	// SHOW MASTER STATUS;
dumpling/export/sql.go:// ShowMasterStatus get SHOW MASTER STATUS result from database
dumpling/export/sql.go:	const showMasterStatusQuery = "SHOW MASTER STATUS"
```

```
2024-05-07T09:53:25.5313474Z [2024/05/07 09:53:25.531 +00:00] [INFO] [dump.go:204] ["get global metadata failed"] [error="sql: SHOW MASTER STATUS: sql: SHOW MASTER STATUS, args: []: Error 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'MASTER STATUS' at line 1"]
```

- [ ] Replace `SHOW MASTER STATUS` with `SHOW BINARY LOG STATUS`
- [ ] Replace `SHOW SLAVE STATUS` with `SHOW REPLICA STATUS`
- [ ] Update CI with MySQL 8.4.0

### PR DESCRIPTION
**Pull request by dveeden**
### What problem does this PR solve?

Issue Number: close #53082 

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [x] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
Dumpling now supports dumpling databases from MySQL 8.4 and newer
```

### CODE REVIEW

File: dumpling/export/sql.go [hunk 1]
```diff
@@ -763,7 +763,17 @@ func ShowMasterStatus(db *sql.Conn) ([]string, error) {
 		}
 		return rows.Scan(addr...)
 	}
-	const showMasterStatusQuery = "SHOW MASTER STATUS"
+
+	// MySQL 8.4.0 and newer: SHOW BINARY LOG STATUS
+	// TiDB, MariaDB, Old MySQL: SHOW MASTER STATUS
+	showMasterStatusQuery := "SHOW MASTER STATUS"
+	if serverInfo.ServerVersion != nil {
+		if !serverInfo.ServerVersion.LessThan(*minNewTerminologyMySQL) &&
```
Comment (lance6716) [inline dumpling/export/sql.go:771]: nit: swap the order, check type is MySQL first

**Review by lance6716 — APPROVED** [review]
/hold

free to unhold

**Comment by lance6716** [comment]
Maybe CI is also needed to be changed

```
++(dumpling/tests/consistency/run.sh:29): run_sql 'show master status;'
ERROR 1064 (42000) at line 1: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'master status' at line 1
```

File: dumpling/export/sql.go (hunk 1 above)
Comment (dveeden) [reply dumpling/export/sql.go]: 9cf43306f3712954d0b8785a807e629e8d8a2507

**Review by GMHDBJD — APPROVED** [review]
LGTM

**Pull request merged**